ast/check.go

```diff
@@ -44,7 +47,24 @@ func (tc *typeChecker) CheckBody(env *TypeEnv, body Body) (*TypeEnv, Errors) {
 		env = env.wrap()
 	}
 
+	prev := len(tc.errs)
 	WalkExprs(body, func(expr *Expr) bool {
+		// If walking this body resulted in errors, then we should not
```

Comment:
This is not by design; it's a bug in the ref checking. We should fix this so that the compiler can return as many useful errors as possible.